osu.Server.Spectator/Hubs/Spectator/SpectatorHub.cs

```diff
@@ -145,10 +146,16 @@ public async Task EndPlaySession(SpectatorState state)
                     if (status < min_beatmap_status_for_replays || status > max_beatmap_status_for_replays)
                         return;
 
-                    score.ScoreInfo.Date = DateTimeOffset.UtcNow;
+                    // if the user never hit anything, further processing that depends on the score existing can be waived because the client won't have submitted the score anyway.
+                    // note that this isn't an early return as we still want to end the play session.
+                    // see: https://github.com/ppy/osu/blob/a47ccb8edd2392258b6b7e176b222a9ecd511fc0/osu.Game/Screens/Play/SubmittingPlayer.cs#L281
```

Comment:
I don't quite understand what this comment means by "end the play session" - there's nothing running after the block other than the finally, right?

Reply:
There is actually:

https://github.com/ppy/osu-server-spectator/blob/cc1d36da977b363d2ac21d93a3b9690f40c26146/osu.Server.Spectator/Hubs/Spectator/SpectatorHub.cs#L159

Skipping that causes a test to fail. I considered moving that call inside `finally` but wasn't convinced that was correct.

Reply:
Oh, it was hidden. How come that isn't an issue for the above `return`?

Reply:
I can understand this one skipping it:

https://github.com/ppy/osu-server-spectator/blob/cc1d36da977b363d2ac21d93a3b9690f40c26146/osu.Server.Spectator/Hubs/Spectator/SpectatorHub.cs#L138-L141

because it assumes that `BeginPlaySession()` failed. But I don't understand this one:

https://github.com/ppy/osu-server-spectator/blob/cc1d36da977b363d2ac21d93a3b9690f40c26146/osu.Server.Spectator/Hubs/Spectator/SpectatorHub.cs#L143-L146

I think that may be an actual bug?

Reply:
Hmmm okay. That second one is the one I was referring to specifically. I think this is fine to get in as-is but that should probably be investigated.

Reply:
Yeah I'm looking at it.